helix-term/src/ui/statusline.rs

```diff
@@ -432,12 +432,14 @@ fn render_file_modification_indicator<F>(context: &mut RenderContext, write: F)
 where
     F: Fn(&mut RenderContext, String, Option<Style>) + Copy,
 {
-    let title = (if context.doc.is_modified() {
+    debug_assert!(!(context.doc.readonly.unwrap_or(false) && context.doc.is_modified()));
```

Comment:
This debug assert is wrong. We have no eadonky editor mode right now. The document can definitly be modified and read-only (even editors that have a read-only mode allow that)